packages/amplify-category-auth/src/provider-utils/awscloudformation/utils/project-has-auth.ts

```diff
@@ -1,15 +1,25 @@
+import { $TSContext, $TSObject, stateManager } from 'amplify-cli-core';
+import { printer } from 'amplify-prompts';
 import { messages } from '../assets/string-maps';
 
 /**
  * Checks if auth already exists in the project and prints a warning if so.
  * Returns true if auth already exists, false otherwise
- * @param context The amplify context
  */
-export const projectHasAuth = (context: any) => {
-  const existingAuth = context.amplify.getProjectDetails().amplifyMeta.auth || {};
+export const projectHasAuth = (context: $TSContext) => {
+  const meta = stateManager.getMeta(undefined, { throwIfNotExist: false });
+  const existingAuth: [string, $TSObject][] = Object.entries(meta?.auth || {});
 
-  if (Object.keys(existingAuth).length > 0) {
-    context.print.warning(messages.authExists);
+  if (existingAuth.length > 0) {
+    if (existingAuth.filter(([_, resource]) => resource?.serviceType === 'imported').length > 0) {
```

Comment:
It would be good to move this into a function checkResourcesWithAuthImported() from a readability perspective.